src/components/BaseModal/BaseModal.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from 'react';
+import './BaseModal.css';
+import BaseCard from '../BaseCard';
+
+interface BaseModalProps {
+  isOpen?: boolean;
+  renderHeader?: React.ReactNode;
+  onClose?: (value: boolean) => void;
+  children?: React.ReactNode;
+  className?: string;
+}
+
+function BaseModal({ isOpen = false, renderHeader, onClose, children, className = '' }: BaseModalProps) {
+  useEffect(() => {
+    document.body.style.overflow = isOpen ? 'hidden' : 'auto';
+  }, [isOpen]);
+
+  return (
+    isOpen && (
+      <div className={`base-modal-container ${className}`}>
+        <div
+          className="base-modal-container__backdrop"
+          onClick={() => onClose?.(!isOpen)}
+          role="button"
+          tabIndex={0}
+          onKeyDown={(e) => {
+            if (e.key === 'Escape') onClose?.(!isOpen);
+          }}
+          aria-label="modal backdrop"
+        />
+        <BaseCard boxShadow={false} className="base-modal-container__modal">
+          {renderHeader && <div className="base-modal-container__modal__header">{renderHeader}</div>}
```

Comment:
Las render props deben ser funciones, siguiendo las reglas del `Render Props pattern`, al ser funciones una de las ventajas que nos da es poder pasarle valores al componente o elemento que queramos renderizar, si bien este no es el caso donde queramos pasar valores hacia arriba, hay que cumplir con el patron y sus reglas.

[Render props](https://www.dhiwise.com/post/leveraging-react-render-props-for-flexible-component-composition#why-use-render-props-pattern-in-react-)

```suggestion
          {renderHeader && <div className="base-modal-container__modal__header">{renderHeader()}</div>}
```

En caso de querer que sea solo texto, sugiero cambiarle el nombre de la prop a otra que no inicie con `render`, por ejemplo `title` o algo por el estilo haciendo referencia al titulo que tendra el modal en su header